### DISCUSSION TITLE
Resolving Transient runtime Error: auto_advance=false

### DISCUSSION THREAD

**Discussion by japresa** [idaholab/moose#18556] [Q&A General] [answered]
Hello,
In my SAM-MOOSE simulation, I'm getting this runtime error and don't know how to fix it.
----------------------
FullSolveMultiApp is not compatible with auto_advance=false
-------------------
I've tried making a SolveStep object and set auto_advance=true but it isn't recognized. I've also tried to manually change the bool value in other blocks but am still getting the error. Any advice on how to get rid of this error would help :)

**Comment by GiudGiud** [comment]
Hello

Have you tried setting that parameter to false to the MultiApps block?
Is another moose app the subapp?

Guillaume

**Comment by japresa** [comment]
Hello

I've tried to manually change the parameter's bool value in the MultiApp block.

What do you mean by "another moose app being the subapp"? I know this subapp, (FullSolveMultiApp), runs a complete simulation every execution? That's the only app I define in the MultiApp block.

**Reply by GiudGiud** [reply]
is the sub app SAM or is it a different app

**Reply by japresa** [reply]
Yes, I’m using SAM

**Comment by GiudGiud** [comment, marked as answer]
I think something in SAM is preventing you from running FullSolveMultiApp, by hard setting auto_advance=false. I just tried on some SAM inputs of mine and got the same error.
@andrsd any thoughts on this?
Do you have the github handles of SAM developers we could tag here?

**Reply by travismui** [reply]
Hi @GiudGiud, I'm part of the SAM developer team. Just wanted to check– the `auto_advance = true` parameter is set in the `Executioner` block, correct? This should be allowed, I don't believe SAM enforces any `Executioner` parameters. I'm not sure if anyone has previously worked with SAM on a `FullSolveMultiApp` problem, but it should be possible. 

I just checked with one of our regression test `TransientMultiApp` problems and was able to run it as a `FullSolveMultiApp` by setting `auto_advance = true` in the `Transient` executioner block of the main app. It seems like it yielded the expected behavior, namely the sub app executing a full simulation for each tilmestep of the main app.

**Reply by GiudGiud** [reply]
Hi Travis

Thanks for pitching in!
Good catch, it's an executioner parameter not a MultiApp parameter.
I was able to run a FullSolveMultiApp in my SAM inputs as well by setting this option. 

SAM (or something) must be enforcing this parameter in some fashion. Other MOOSE apps do not require setting this parameter before using a FullSolveMultiApp.

Guillaume
